### PR TITLE
add target cpu info to configuration

### PR DESCRIPTION
**Pull request by Tieske**
- site_config remains local, no longer exposed
- added field 'target_cpu' to `cfg` and config file environments

as discussed in https://github.com/keplerproject/luarocks/issues/390#issuecomment-114852615

### CODE REVIEW

File: src/luarocks/cfg.lua [hunk 1]
```diff
@@ -630,4 +630,13 @@ function cfg.is_platform(query)
...
 
+--[[
+function cfg.dump()
+  for k,v in pairs(defaults) do
+    if cfg[k] then end -- nothing todo, just touch it and the __index will make a local copy in `cfg` 
+  end
+  print(util.show_table(cfg, "Dump of `cfg` table:"))
+end
+cfg.dump() --]]
+
```
Comment (hishamhm) [inline src/luarocks/cfg.lua]: not a fan of leaving debug stuff around as comments like that...

**Comment by Tieske** [comment]
oops... fixed!

**Comment by ignacio** [comment]
I had a go at testing this but I think something is wrong. luarocks.cfg seems to be overriding my config file with wrong values.

This is what the installer found:
~~~~
==========================
== System check results ==
==========================

Will configure LuaRocks with the following paths:
LuaRocks        : c:\luarocks\2.2
Config file     : c:\luarocks\config.lua
Rocktree        : c:\luarocks\systree

Lua interpreter : c:\lua\X64\5.1.5\bin\\lua.exe
    binaries    : c:\lua\X64\5.1.5\bin\
    libraries   : c:\lua\X64\5.1.5\lib\
    includes    : c:\lua\X64\5.1.5\include\
    architecture: x86_64
    binary link : lua51.lib with runtime MSVCR120.dll
~~~~

This is part of my config file:
~~~~
variables = {
    MSVCRT = 'MSVCR120',
    LUALIB = 'lua51.lib'
}
~~~~

And it seems that https://github.com/keplerproject/luarocks/blob/master/src/luarocks/cfg.lua#L342 is overriding it with the wrong value.

**Comment by ignacio** [comment]
Ok... nevermind, I saw that "remains; configuring targets from config.lua". Until that is in place, not even the builtin build backend works on Windows. (MSVCRT variable is nil)

**Comment by Tieske** [comment]
No... you're right. But the problem was introduced in #385. The config file was directly loaded into `cfg`, which exposed `cfg` to the config file. After changing that, the content of the config file was no longer merged into `cfg`, because that should now be an extra step.

Fixed it, PR is upcoming

**Comment by Tieske** [comment]
see PR #397 

please give this another try, it should also allow updating targets if you add `target_cpu = x86_64` in `config.lua` for example.

**Comment by ignacio** [comment]
Thanks. Will check that first thing tomorrow morning.

**Comment by ignacio** [comment]
@Tieske Just checked and it is working fine. Also overriding target_cpu on the config file. The PR is good to merge.

Thanks!

**Pull request merged**